pkg/api/metrics.go

```diff
@@ -44,6 +34,38 @@ func (hs *HTTPServer) QueryMetricsV2(c *models.ReqContext) response.Response {
 	return toJsonStreamingResponse(resp)
 }
 
+func (hs *HTTPServer) handleQueryMetricsError(err error) *response.NormalResponse {
+	if errors.Is(err, models.ErrDataSourceAccessDenied) {
+		return response.Error(http.StatusForbidden, "Access denied to data source", err)
+	}
+	if errors.Is(err, models.ErrDataSourceNotFound) {
+		return response.Error(http.StatusNotFound, "Data source not found", err)
+	}
+	var badQuery *query.ErrBadQuery
+	if errors.As(err, &badQuery) {
+		return response.Error(http.StatusBadRequest, util.Capitalize(badQuery.Message), err)
+	}
+
+	if errors.Is(err, backendplugin.ErrPluginNotRegistered) {
+		return response.Error(http.StatusNotFound, "Plugin not found", err)
+	}
+
+	if errors.Is(err, backendplugin.ErrMethodNotImplemented) {
+		return response.Error(http.StatusBadRequest, "Plugin functionality not implemented", err) // or 404
+	}
+
+	if errors.Is(err, backendplugin.ErrPluginUnavailable) {
+		return response.Error(http.StatusBadGateway, "Plugin unavailable", err)
+	}
+
+	var failedQuery *backendplugin.ErrFailedQuery
```

Comment:
This could be 200/207 (if we decide to go that path) since an error originating from a plugin.